test/units/testsuite-73.sh

```diff
@@ -92,6 +92,19 @@ test_locale() {
         return
     fi
 
+    # start with a known default environment and make sure to also give a
+    # default value to LC_CTYPE= since we're about to also set/unset it. We
+    # also reload PID1 configuration to make sure that PID1 environment itself
+    # is updated as it's not always been the case.
+    assert_rc 0 localectl set-locale "LANG=en_US.UTF-8" "LC_CTYPE=C"
+    systemctl daemon-reload
```

Comment:
So, this can be dropped after the second commit, right?

Reply:
In theory yes `daemon-reload` is implied by `set-locale` with the second commit. But I would still let it in place just in case it will be broken again in the future.

Reply:
we want to notice if it gets broken, no?

Reply:
That should be trapped by the code that follows `daemon-reload`, where `LC_CTYPE` is set then unset and PID1 environment is checked with `systemctl show-environment`.

`daemon-reload` here is used to make sure that PID1 reads /etc/locale.conf and initializes its "transient" environment  (the one PID1 read from its config files) in a known state. This way after unsetting `LC_CTYPE` and checking that it's no more present in PID1 environment we are sure that this env variable was dropped (since it was initially part of the env) from the transient environment.